presto-main/src/main/java/com/facebook/presto/execution/DistributedNodeTaskMap.java

```diff
@@ -0,0 +1,124 @@
+import com.facebook.presto.metadata.InternalNode;
+import com.facebook.presto.resourcemanager.ForResourceManager;
+import com.facebook.presto.resourcemanager.NodeTaskState;
+import com.facebook.presto.resourcemanager.ResourceManagerClient;
+import com.facebook.presto.resourcemanager.ResourceManagerConfig;
+import com.facebook.presto.util.PeriodicTaskExecutor;
+import com.google.common.collect.ImmutableMap;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import javax.inject.Inject;
+
+import java.util.Map;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static java.util.Objects.requireNonNull;
+
+public class DistributedNodeTaskMap
+        implements NodeTaskMap
+{
+    public static final NodeTaskState EMPTY_NODE_TASK_STATE = new NodeTaskState(PartitionedSplitsInfo.forZeroSplits(), 0, 0);
+    private final DriftClient<ResourceManagerClient> resourceManagerClient;
+    private final long nodeTaskMapFetchIntervalMillis;
+    private final PeriodicTaskExecutor nodeTaskMapUpdater;
+
+    private final AtomicReference<Map<String, NodeTaskState>> nodeTaskStateMap;
+
+    @Inject
+    public DistributedNodeTaskMap(
+            @ForResourceManager DriftClient<ResourceManagerClient> resourceManagerClient,
+            @ForResourceManager ScheduledExecutorService executorService,
+            ResourceManagerConfig resourceManagerConfig)
+    {
+        this.resourceManagerClient = requireNonNull(resourceManagerClient, "resourceManagerClient is null");
+        this.nodeTaskMapFetchIntervalMillis = requireNonNull(resourceManagerConfig, "resourceManagerConfig is null").getNodeTaskMapFetchInterval().toMillis();
+        nodeTaskStateMap = new AtomicReference<>(ImmutableMap.of());
+        this.nodeTaskMapUpdater = new PeriodicTaskExecutor(nodeTaskMapFetchIntervalMillis, executorService, () -> nodeTaskStateMap.set(updateNodeTaskStateMap()));
+    }
+
+    @PostConstruct
+    public void init()
+    {
+        nodeTaskMapUpdater.start();
+    }
+
+    @PreDestroy
+    public void stop()
+    {
+        nodeTaskMapUpdater.stop();
+    }
+
+    @Override
+    public void addTask(InternalNode node, RemoteTask task)
+    {
+        // no op
+    }
+
+    @Override
+    public PartitionedSplitsInfo getPartitionedSplitsOnNode(InternalNode node)
+    {
+        return nodeTaskStateMap.get().getOrDefault(node.getNodeIdentifier(), EMPTY_NODE_TASK_STATE).getPartitionedSplitsInfo();
+    }
+
+    @Override
+    public long getNodeTotalMemoryUsageInBytes(InternalNode node)
+    {
+        return nodeTaskStateMap.get().getOrDefault(node.getNodeIdentifier(), EMPTY_NODE_TASK_STATE).getTotalMemoryUsageInBytes();
+    }
+
+    @Override
+    public double getNodeCpuUtilizationPercentage(InternalNode node)
+    {
+        return nodeTaskStateMap.get().getOrDefault(node.getNodeIdentifier(), EMPTY_NODE_TASK_STATE).getCpuUtilizationPercentage();
```

Comment:
Similarly, this will return process CPU load, which is the CPU utilization of the JVM process, including internal threads. That's different from LocalNodeTaskMap's implementation, which tracks just task CPU.